csharp/Adapter/Microsoft.Spark.CSharp/Network/RioNative.cs

```diff
@@ -128,25 +129,34 @@ private void Init()
             var maxThreads = 1;
             if (useThreadPool)
             {
-                maxThreads = Environment.ProcessorCount;
+                maxThreads = (Environment.ProcessorCount + 1) >> 1;
```

Comment:
How will this calculation for maxThreads work within a YARN container where it is possible to control the number of cores?